src/pages/auth/Signup.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { useRef, useState, useEffect } from "react";
+import "./auth.css";
+
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import {
+  validateName,
+  validateEmail,
+  validatePassword,
+} from "./utils/inputValidateUtils";
+import { useAuth } from "../../context";
+import { useAxios } from "../../utils/useAxios";
+
+const SignUp = () => {
+  const initalFormValidateState = {
```

Comment:
You can move this object out of the component body. That will it won't be recreated on each re render